File: bench/memory/1.cpp
#include <iostream>
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
	int S = 1000000000; // 1 GB
	if (argc == 2)
	{
		S = atoi(argv[1]);
	}

	S = S / sizeof(int);

	auto start = chrono::steady_clock::now();

	int *p = (int *)malloc(S * sizeof(int));
	for (int i = 0; i < S; i++)
	{
		p[i] = 42;
	}

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	free(p);

	return 0;
}

File: bench/memory/1.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	var S = 1_000_000_000 // 1 GB
	var args = os.Args[1:]
	if len(args) >= 1 {
		S, _ = strconv.Atoi(args[0])
	}

	S = S / strconv.IntSize

	var start = time.Now()

	var p = make([]int, S)
	for i := range p {
		p[i] = 42
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}
